server/db/models/player.js

```diff
@@ -148,204 +89,183 @@ playerSchema.methods.setCardinal = function(row, col) {
-        magnitude --;
-      }
-      this.set('position', [newRow, newCol]);
-    }
-  })
-}
 
+    this.position = [newRow, newCol]
+
+    this.touchFlag(newRow, newCol)
+  }
+}
 
 
 playerSchema.methods.checkForPit = function(row, col) {
-  return this.findMyTile(row, col)
-  .then(function(tile) {
-    if (tile.floor === 'pit') return true;
-    else return false;
-  })
+  var tile = this.game.getTileAt(row, col)
+  if (tile.floor === 'pit') return true;
+  else return false;
 }
 
+
 playerSchema.methods.touchFlag = function(row, col) {
-  var player = this;
-  return player.findMyTile(row, col)
-  .then(function(tile) {
-    if (tile.flag != null) {
-      if (player.flagCount + 1 === tile.flag) {
-        player.flagCount++;
-        return player.save()
-      }
+  var tile = this.game.getTileAt(row, col)
+  if(tile.flag != null) {
+    if(this.flagCount + 1 === tile.flag) {
+      this.flagCount++;
+      this.checkIfWinner()
     }
-  })
-  .then(function(savedPlayer) {
-    return savedPlayer.checkIfWinner()
-  })
-  .then(function(result) {
-    if (result === true) return player.gameover()
-  })
+  }
 }
 
+
 playerSchema.methods.checkIfWinner = function() {
-  var player = this;
-  return mongoose.model('Game').findById(player.game)
-  .then(function(game) {
-    if (game.numFlags === player.flagCount) return true;
-    else return false;
-  })
+  if (this.game.numFlags === this.flagCount) this.gameover();
+  else return;
 }
 
-playerSchema.methods.applyDamage = function(hitCount){
-  var player = this;
-  return player.update({$inc: {damage: hitCount}}, {new: true})
-  .then(function() {
-    return player.checkDamage()
-  })
+
+playerSchema.methods.gameover = function() {
+  this.game.state = 'gameover'
+  //at this point, we should stop everything and push game state into gameFB array
+  //potentially save to database?
 };
 
-playerSchema.methods.checkDamage = function() {
-  if (this.damage > 9) {
-    this.set('damage', 0)
-    this.save()
-    .then(function(player) {
-      player.loseLife()
-    })
-  }
+
+playerSchema.iAmReady = function() {
+  this.ready = true
 }
 
-playerSchema.methods.loseLife = function() {
-  this.livesRemaining--;
-  if (this.livesRemaining === 0) return this.killPlayer();
-  else {
-    this.set('position', this.dock);
-    this.save();
-  }
-};
 
-playerSchema.methods.killPlayer = function() {
-  this.set('_id', null); //how should we kill players?
-  this.save();
+playerSchema.methods.isPlayerReady = function() {
+    return this.ready;
 };
 
-playerSchema.methods.gameover = function() {
-  return mongoose.model('Game').findById(this.game)
-  .then(function(game) {
-    game.state = 'gameover';
-    game.save();
-  });
-};
 
-playerSchema.methods.findMyTile = function(row, col){
-  var player = this;
-  var row = row || player.position[0];
-  var col = col || player.postion[1];
+playerSchema.methods.evaluateDamage = function(hitCount) {
+  this.applyDamage(hitCount)
+  this.checkDamage()
+}
```

Comment:
could we call this apply damage and the other one something else like accrue damage?